internal/config/nsmgr/types_unsupported.go

```diff
@@ -0,0 +1,28 @@
+//go:build !linux
+// +build !linux
```

Comment:
is there a way we can avoid this file? I see a lot of the same content as types_linux, so maybe we can take the types.go content and put any platform specific pieces in nsmgr_*

Reply:
In this PR, I was aiming to blindly rename things to make it easier to review. I can add a separate commit to this PR which factor out the common parts though.

Reply:
I am fine with a slightly harder review--I think I'd prefer it squashed with the existing commits (but maintaining the different commits per directory)

Reply:
Looking at the freebsd version of types, the sharable parts are mainly the bits that I moved to types_unsupported.go so I'll re-work this commit to leave those bits in types.go.